cdk/kraken/src/deploy.ts

```diff
@@ -37,50 +43,77 @@ export class DeployJob extends CheckoutJob {
     const fullConfig: Required<DeployJobProps> = {
       deployTag: "${{ github.sha }}",
       defaultBranch: "master",
+      isFeatureDeploy: false,
       ...config,
     };
 
-    super(scope, "deploy", {
-      runsOn: "ubuntu-latest",
-      if: `github.ref == 'refs/heads/${fullConfig.defaultBranch}'`,
-      steps: [
-        {
-          id: "synth",
-          name: "Synth cdk8s manifests",
-          run: dedent`cd k8s
+    super(
+      scope,
+      fullConfig.isFeatureDeploy ? "feature-branch-deploy" : "deploy",
+      {
+        runsOn: "ubuntu-latest",
+        if: fullConfig.isFeatureDeploy
+          ? `startsWith(github.ref, 'refs/heads/feature/') == true`
+          : `github.ref == 'refs/heads/${fullConfig.defaultBranch}'`,
+        steps: [
+          ...(fullConfig.isFeatureDeploy
+            ? [
+                {
+                  name: "Get Pull Request Number",
+                  id: "pr",
+                  run: dedent`
+          echo "::set-output name=pull_request_number::$(gh pr view --json number -q .number || echo "")"
+                  `,
+                  env: {
+                    GITHUB_TOKEN: "${{ secrets.GITHUB_TOKEN }}",
+                  },
+                },
+              ]
+            : []),
```

Comment:
These spreads are looking a little messy :'D but this is the best way imo to go about it and what the developer sees is very clean so 👍 yay 😔